### PR TITLE
rust/rfb: use consistent key name for `security_result`

### PR DESCRIPTION
**Pull request by satta**
- [X] I have read the contributing guide lines at
   https://docs.suricata.io/en/latest/devguide/contributing/contribution-process.html
- [X] I have signed the Open Information Security Foundation contribution agreement at
   https://suricata.io/about/contribution-agreement/ (note: this is only required once)
- [X] I have updated the user guide (in doc/userguide/) to reflect the changes made (if applicable)

Link to ticket: https://redmine.openinfosecfoundation.org/issues/7198

Describe changes:
- Ensure consistent key `rfb.authentication.security_result`.
- Ensure that this key only is mentioned in the documentation.

### CODE REVIEW

**Comment by victorjulien** [comment]
I wonder if we should make a mention of this in the upgrade guide. It may break some ppls expectations if they relied on the old inconsistent behavior. No-one complained to us though, so probably not a big issue.

**Comment by satta** [comment]
> I wonder if we should make a mention of this in the upgrade guide. It may break some ppls expectations if they relied on the old inconsistent behavior. No-one complained to us though, so probably not a big issue.

I'd say a quick mention can't hurt. So you are suggesting that we keep it for 7.0.x and just make it part of the transition from 7 to 8?

**Comment by catenacyber** [comment]
> > I wonder if we should make a mention of this in the upgrade guide. It may break some ppls expectations if they relied on the old inconsistent behavior. No-one complained to us though, so probably not a big issue.
> 
> I'd say a quick mention can't hurt. So you are suggesting that we keep the old behaviour for 7.0.x and just make it part of the transition from 7 to 8?

I vote for this PR rather than status quo for suricata 7

File: doc/userguide/output/eve/eve-json-format.rst [hunk 1]
```diff
@@ -1895,7 +1895,7 @@ Fields
 * "client_protocol_version.major", "client_protocol_version.minor": The RFB protocol version agreed by the client.
 * "authentication.security_type": Security type agreed upon in the logged transaction, e.g. ``2`` is VNC auth.
 * "authentication.vnc.challenge", "authentication.vnc.response": Only available when security type 2 is used. Contains the challenge and response byte buffers exchanged by the server and client as hex strings.
-* "authentication.security-result": Result of the authentication process (``OK``, ``FAIL`` or ``TOOMANY``).
+* "authentication.security_result": Result of the authentication process (``OK``, ``FAIL`` or ``TOOMANY``).
```
Comment (catenacyber) [inline doc/userguide/output/eve/eve-json-format.rst:1898]: Is this to be changed in master as well ?

Comment (satta) [reply doc/userguide/output/eve/eve-json-format.rst:1898]: Yes, it's also an existing inconsistency in the documentation.

Comment (catenacyber) [reply doc/userguide/output/eve/eve-json-format.rst:1898]: So, could you fix the documentation in master first please ? (then cherry-pick the doc fix + add a new commit to fix the code in 7)

Comment (satta) [reply doc/userguide/output/eve/eve-json-format.rst:1898]: Sure. PR to follow.

**Pull request closed without merging**